### DISCUSSION TITLE
Understanding Camera and Controller

### DISCUSSION THREAD

**Discussion by Kickbykick** [Sixze/ALS-Refactored#65] [Q&A]
Hi there,

I am trying to understand how the Controller and Camera works. I know that you can connect a Controller to a CameraManager but the Camera system you have in this repo is not APlayerCameraManager so what is the connection.

Can you please briefly explain how the Controller knows that it should use the AlsCamera system that you implemented. I don't see any possession of the camera or any use of the camera in the controller but some how the controller controls the camera.

Thanks for the help.

**Comment by Sixze** [comment]
Camera is implemented as a component similar to the standard camera component, no need for custom `APlayerCameraManager` or `APlayerController` classes, but additional setup is required in the C++ character class and character blueprint (see `AAlsCharacterExample` class and `BP_Als_Character` blueprint).

**Reply by Kickbykick** [reply]
```
virtual void CalcCamera
(
    float DeltaTime,
    struct [FMinimalViewInfo](https://docs.unrealengine.com/4.26/en-US/API/Runtime/Engine/Camera/FMinimalViewInfo/index.html) & OutResult
)
```

Looked through the entire code and I think this does it. This function is `APlayerController::CalcCamera` so it can affect the player controller and by turn affecting the camera. Is this observation right?

I have read through most of the code and I understand it, I have just been looking for that one function that talks to the player controller and I think this is it. Because in my head I was like if I have multiple camera components, what tells unreal engine that this is what you should look at.
Let me know if this observation is right tho.

**Comment by Sixze** [comment]
`APlayerController::CalcCamera()` is used only if there is no pawn. In all other cases, `AAlsCharacterExample::CalcCamera()` is used instead.
